cx2t/src/claw/wani/transformation/ll/loop/LoopFission.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is released under terms of BSD license
+ * See LICENSE file for more information
+ */
+package claw.wani.transformation.ll.loop;
+
+import claw.shenron.transformation.Transformation;
+import claw.shenron.translator.Translator;
+import claw.tatsu.primitive.Loop;
+import claw.tatsu.xcodeml.exception.IllegalTransformationException;
+import claw.tatsu.xcodeml.xnode.common.XcodeProgram;
+import claw.tatsu.xcodeml.xnode.common.Xcode;
+import claw.tatsu.xcodeml.xnode.common.XcodeML;
+import claw.tatsu.xcodeml.xnode.common.Xnode;
+import claw.wani.language.ClawPragma;
+import claw.wani.transformation.ClawTransformation;
+
+/**
+ * A LoopFission transformation is an independent transformation that can
+ * be used to split a large loop into multiple smaller ones. If the
+ * proposed split violates data dependencies within the original loop
+ * body, the transformation will attempt to promote the according variables
+ * to maintain the correctness of the code.
+ *
+ * Please note that this is still experimental at this stage!
+ *
+ * @author mlange05
+ */
+
+public class LoopFission extends ClawTransformation {
+
+  // The do loop that we want to split
+  private Xnode _loop;
+  private Xnode _marker;
+
+  /**
+   * Constructs a new LoopFission triggered from a specific pragma.
+   *
+   * @param directive The directive that triggered the loop fission
+   *                  transformation.
+   */
+  public LoopFission(ClawPragma directive) throws IllegalTransformationException
+    {
+    super(directive);
+
+    this._marker = directive.getPragma();
+    this._loop = this._marker.ancestor().ancestor();
```

Comment:
@mlange05 I guess you have mixed up loops in the multiple fission example because you have to re-analyze this before transforming as it might have changed since the transformation instance creation.